RIPS/rip-7560.md

```diff
@@ -241,47 +242,54 @@ The amount of gas used by this frame is referred to as `senderValidationGasUsed`
 
 The frame must return 32 bytes `validationData` that is interpreted as:
 
-```solidity
-MAGIC_VALUE_SENDER <20 bytes>, validUntil <6 bytes>, validAfter <6 bytes>
-```
+- **authorizer** (20 bytes) - (magic value/address, defined below)
+- **validUntil** (6 bytes) - a timestamp, or zero for "infinite". The transaction is valid only up to this time.
+- **validAfter** (6 bytes) - a timestamp. The transaction is valid only after this time.
 
-In order to allow a gas estimation of the transaction prior signing, this function
-should avoid reverting on invalid signature, and return `MAGIC_VALUE_SIGFAIL` instead.
+Currently, **authorizer** is treated as a magic value:
+- `MAGIC_VALUE_SENDER` - returned by the account after it verified the transaction (and agrees to pay for its execution)
+- `MAGIC_VALUE_SIGFAIL` - returned by the account if the transaction is syntactically valid, but the signature is not.
+    This valid is used during gas estimation, and it helps wallet and apps to determine the gas used by the validation process.
+- Any other returned value is considered failure, and the transaction is rejected.
+- Future protocol MAY define authorizer as an on-chain contract to perform further account authorization, such as aggregation
 
-Type of the `validUntil` is 6-byte timestamp value, or zero for "infinite". The transaction is valid only up to this time.
-Type of the `validAfter` is 6-byte timestamp. The transaction is valid only after this time.
-
-The `validateTransaction` function can choose to revert on any condition that can be satisfied during gas estimation.
+A revert by `validateTransaction` indicates rejection of the transaction, and it is not included on-chain.
 
 #### Paymaster validation frame
 
 The `paymaster` of the transaction, if specified, is invoked with the following data:
 
 ```solidity
 
-function validatePaymasterTransaction(uint256 version, bytes32 txHash, bytes transaction) external returns (bytes32 validationData, bytes context);
+function validatePaymasterTransaction(uint256 version, bytes32 txHash, bytes transaction) 
+external returns (bytes32 validationData, bytes context);
 
 ```
 
 The gas limit of this frame is set to `paymasterValidationGasLimit`.
 
 The amount of gas used by this frame is referred to as `paymasterValidationGasUsed`.
 
-The `transaction` parameter is interpreted as an ABI encoding of `TransactionType4`.\
-The `txHash` parameter represents the hash of the AA_TX_TYPE transaction with empty signature, as defined in section
+- The `version` parameter is `AA_TX_TYPE`
```

Comment:
No, it isn't. The `version` is a mechanism to change the structure without necessarily changing the transaction type. This mechanism is still TBD though.

Reply:
then suggest a fix to this line - either remove it altogether, or suggest another value (e.g. AA_TX_VERSION?)